### DISCUSSION TITLE
Help me. Unable to get real time data from database

### DISCUSSION THREAD

**Discussion by aktweb** [vercel/next.js#61245] [Help]
### Summary

this is my Project , where in i have a table which is being populated from xata.io database .

https://github.com/aktweb/vak/tree/main/app/dashboard
[https://github.com/aktweb/vak/blob/main/app/dashboard/page.tsx](url)

the problem is  :  The table data is not changing in real time according to database. It is diplaying the data which is fetched while building ,and after deployment the table is not up to date with the database and shows the old data fetched while build , ( I have checked the DB response and it is correct) .

I have been stuck with this problem from so long, it would be great if you could answer this. thank you

**Comment by nirvikpurkait** [comment]
When you are deploying your application, your application gets builds first, and when building if you fetch some data (using fetch or database call), the data gets cached, component renders statically, and the data always stays the same.

If you want to update the data you can make use of some functions [`unstable_noStore`](https://nextjs.org/docs/app/api-reference/functions/unstable_noStore) or [`cookies`](https://nextjs.org/docs/app/api-reference/functions/cookies) that makes the rendering dynamic (on demand render), if you want to update the data data and refelect changes instantly, you can make use of `revalidateTag` or `revalidatePath`
